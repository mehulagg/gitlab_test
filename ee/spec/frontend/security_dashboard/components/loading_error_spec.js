import { shallowMount } from '@vue/test-utils';

import { GlEmptyState } from '@gitlab/ui';

import LoadingError from 'ee/security_dashboard/components/loading_error.vue';

describe('LoadingError component', () => {
  let wrapper;

  const createWrapper = errorCode => {
    wrapper = shallowMount(LoadingError, {
      propsData: {
        errorCode,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  describe.each([401, 403])('with error code %s', errorCode => {
    beforeEach(() => {
      createWrapper(errorCode);
    });

    it('renders an empty state', () => {
      expect(wrapper.find(GlEmptyState).exists()).toBe(true);
    });

    it('empty state has correct props', () => {
      expect(wrapper.find(GlEmptyState).props()).toMatchSnapshot();
    });
  });
});
