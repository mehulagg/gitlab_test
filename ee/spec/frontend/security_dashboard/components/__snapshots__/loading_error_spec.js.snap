// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingError component with error code 401 empty state has correct props 1`] = `
Object {
  "compact": false,
  "description": "Security reports can only be accessed by authorized users.",
  "primaryButtonLink": "/users/sign_in",
  "primaryButtonText": "Sign in",
  "secondaryButtonLink": null,
  "secondaryButtonText": null,
  "svgPath": "/assets/illustrations/user-not-logged-in.svg",
  "title": "You must sign in as an authorized user to see this report",
}
`;

exports[`LoadingError component with error code 403 empty state has correct props 1`] = `
Object {
  "compact": false,
  "description": "Security reports can only be accessed by authorized users.",
  "primaryButtonLink": null,
  "primaryButtonText": null,
  "secondaryButtonLink": null,
  "secondaryButtonText": null,
  "svgPath": "/assets/illustrations/lock_promotion.svg",
  "title": "You do not have sufficient permissions to access this report",
}
`;
